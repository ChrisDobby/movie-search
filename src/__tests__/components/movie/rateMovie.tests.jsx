import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import RateMovie from '../../../components/movie/rateMovie';
import RatingValue from '../../../components/movie/ratingValue';

jest.mock('../../../components/withAuthentication');

describe('rateMovie', () => {
    const defaultProps = {
        id: 1,
        isAuthenticated: true,
        authenticate: jest.fn(() => Promise.resolve({})),
        actions: {},
    };

    it('should set the rating in state when setRating called', () => {
        const page = shallow(<RateMovie {...defaultProps} />).dive();

        page.instance().setRating(7);

        expect(page.state().rating).toBe(7);
    });

    it('should set the rating when rating component selected', () => {
        const page = shallow(<RateMovie {...defaultProps} />).dive();
        const ratingValue = page.find(RatingValue).at(0);

        ratingValue.props().select();

        expect(page.state().rating).toBe(ratingValue.props().value);
    });

    it('should render correctly when authenticating', () => {
        const props = {
            ...defaultProps,
            isAuthenticated: false,
            isAuthenticating: true,
        };

        const page = renderer.create(<RateMovie {...props} />);

        expect(page.toJSON()).toMatchSnapshot();
    });

    it('should render correctly after authentication error', () => {
        const props = {
            ...defaultProps,
            isAuthenticated: false,
            authenticationError: true,
        };

        const page = renderer.create(<RateMovie {...props} />);

        expect(page.toJSON()).toMatchSnapshot();
    });

    it('should render correctly when not authenticated', () => {
        const props = {
            ...defaultProps,
            isAuthenticated: false,
        };

        const page = renderer.create(<RateMovie {...props} />);

        expect(page.toJSON()).toMatchSnapshot();
    });

    it('should render correctly when authenticated', () => {
        const props = {
            ...defaultProps,
            isAuthenticated: true,
        };

        const page = renderer.create(<RateMovie {...props} />);

        expect(page.toJSON()).toMatchSnapshot();
    });
});
