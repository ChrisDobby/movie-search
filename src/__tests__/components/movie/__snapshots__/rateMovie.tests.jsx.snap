// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rateMovie should render correctly after authentication error 1`] = `
<div>
  <h4>
    Rate this movie
  </h4>
  <div
    className="alert alert-danger"
    role="alert"
  >
    There was an error logging in to MovieDb.
  </div>
  <button
    className="btn btn-link"
    onClick={[MockFunction]}
    style={
      Object {
        "whiteSpace": "normal",
      }
    }
  >
    You are not logged in to MovieDb.  Click here to login.
  </button>
</div>
`;

exports[`rateMovie should render correctly when authenticated 1`] = `
<div>
  <h4>
    Rate this movie
  </h4>
  <div>
    <div
      className="row"
    >
      <div
        className="col-xs-3 col-sm-2 col-md-2 col-lg-1"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
        style={
          Object {
            "background": "#ffffff",
            "border": "2px solid #ffeeba",
            "borderRadius": "50%",
            "color": "#856404",
            "cursor": "pointer",
            "fontSize": "32px",
            "margin": "4px",
            "maxHeight": "80px",
            "maxWidth": "80px",
            "padding": "8px",
            "textAlign": "center",
          }
        }
      >
        1
      </div>
      <div
        className="col-xs-3 col-sm-2 col-md-2 col-lg-1"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
        style={
          Object {
            "background": "#ffffff",
            "border": "2px solid #ffeeba",
            "borderRadius": "50%",
            "color": "#856404",
            "cursor": "pointer",
            "fontSize": "32px",
            "margin": "4px",
            "maxHeight": "80px",
            "maxWidth": "80px",
            "padding": "8px",
            "textAlign": "center",
          }
        }
      >
        2
      </div>
      <div
        className="col-xs-3 col-sm-2 col-md-2 col-lg-1"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
        style={
          Object {
            "background": "#ffffff",
            "border": "2px solid #ffeeba",
            "borderRadius": "50%",
            "color": "#856404",
            "cursor": "pointer",
            "fontSize": "32px",
            "margin": "4px",
            "maxHeight": "80px",
            "maxWidth": "80px",
            "padding": "8px",
            "textAlign": "center",
          }
        }
      >
        3
      </div>
      <div
        className="col-xs-3 col-sm-2 col-md-2 col-lg-1"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
        style={
          Object {
            "background": "#ffffff",
            "border": "2px solid #ffeeba",
            "borderRadius": "50%",
            "color": "#856404",
            "cursor": "pointer",
            "fontSize": "32px",
            "margin": "4px",
            "maxHeight": "80px",
            "maxWidth": "80px",
            "padding": "8px",
            "textAlign": "center",
          }
        }
      >
        4
      </div>
      <div
        className="col-xs-3 col-sm-2 col-md-2 col-lg-1"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
        style={
          Object {
            "background": "#ffffff",
            "border": "2px solid #ffeeba",
            "borderRadius": "50%",
            "color": "#856404",
            "cursor": "pointer",
            "fontSize": "32px",
            "margin": "4px",
            "maxHeight": "80px",
            "maxWidth": "80px",
            "padding": "8px",
            "textAlign": "center",
          }
        }
      >
        5
      </div>
      <div
        className="col-xs-3 col-sm-2 col-md-2 col-lg-1"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
        style={
          Object {
            "background": "#ffffff",
            "border": "2px solid #ffeeba",
            "borderRadius": "50%",
            "color": "#856404",
            "cursor": "pointer",
            "fontSize": "32px",
            "margin": "4px",
            "maxHeight": "80px",
            "maxWidth": "80px",
            "padding": "8px",
            "textAlign": "center",
          }
        }
      >
        6
      </div>
      <div
        className="col-xs-3 col-sm-2 col-md-2 col-lg-1"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
        style={
          Object {
            "background": "#ffffff",
            "border": "2px solid #ffeeba",
            "borderRadius": "50%",
            "color": "#856404",
            "cursor": "pointer",
            "fontSize": "32px",
            "margin": "4px",
            "maxHeight": "80px",
            "maxWidth": "80px",
            "padding": "8px",
            "textAlign": "center",
          }
        }
      >
        7
      </div>
      <div
        className="col-xs-3 col-sm-2 col-md-2 col-lg-1"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
        style={
          Object {
            "background": "#ffffff",
            "border": "2px solid #ffeeba",
            "borderRadius": "50%",
            "color": "#856404",
            "cursor": "pointer",
            "fontSize": "32px",
            "margin": "4px",
            "maxHeight": "80px",
            "maxWidth": "80px",
            "padding": "8px",
            "textAlign": "center",
          }
        }
      >
        8
      </div>
      <div
        className="col-xs-3 col-sm-2 col-md-2 col-lg-1"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
        style={
          Object {
            "background": "#ffffff",
            "border": "2px solid #ffeeba",
            "borderRadius": "50%",
            "color": "#856404",
            "cursor": "pointer",
            "fontSize": "32px",
            "margin": "4px",
            "maxHeight": "80px",
            "maxWidth": "80px",
            "padding": "8px",
            "textAlign": "center",
          }
        }
      >
        9
      </div>
      <div
        className="col-xs-3 col-sm-2 col-md-2 col-lg-1"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
        style={
          Object {
            "background": "#ffffff",
            "border": "2px solid #ffeeba",
            "borderRadius": "50%",
            "color": "#856404",
            "cursor": "pointer",
            "fontSize": "32px",
            "margin": "4px",
            "maxHeight": "80px",
            "maxWidth": "80px",
            "padding": "8px",
            "textAlign": "center",
          }
        }
      >
        10
      </div>
    </div>
  </div>
</div>
`;

exports[`rateMovie should render correctly when authenticating 1`] = `
<div>
  <h4>
    Rate this movie
  </h4>
  <div
    className="col-md-12"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <div>
      <i
        className="fa fa-spinner fa-spin"
        style={
          Object {
            "fontSize": "24px",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`rateMovie should render correctly when not authenticated 1`] = `
<div>
  <h4>
    Rate this movie
  </h4>
  <button
    className="btn btn-link"
    onClick={[MockFunction]}
    style={
      Object {
        "whiteSpace": "normal",
      }
    }
  >
    You are not logged in to MovieDb.  Click here to login.
  </button>
</div>
`;
